### PR TITLE
Set account nickname to ENS when ENS is available

### LINKED ISSUE #11273: Show ENS reverse-resolved name on account detail view
If a user has not changed their account nickname (it is not `\Account $[0-9]+\`), and an ENS reverse-resolved name is available for that account, then show the reverse-resolved name for that account.

### PR DESCRIPTION
**Pull request by hsy822**
I have a problem for previous PR(#11345 )  and I'm requesting it again :)

Fixes: #11273

Explanation: If the user has not changed the account nickname and ENS is available, ENS reverse-resolved name is displayed on Account detail view instead of the nickname such as 'Account 1'.

Manual testing steps:

Import the account registered in ENS to Metamask
Set the alias of the account to a default value such as 'Account 1'
Go to the Account detail view and check if the ENS value is loaded.

### CODE REVIEW

File: ui/components/app/modals/account-details-modal/account-details-modal.component.js [hunk 1]
```diff
@@ -2,11 +2,18 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { getAccountLink } from '@metamask/etherscan-link';
 
+import { ethers } from 'ethers';
 import AccountModalContainer from '../account-modal-container';
 import QrView from '../../../ui/qr-code';
 import EditableLabel from '../../../ui/editable-label';
 import Button from '../../../ui/button';
 
+import ENS from '../../../../../app/scripts/controllers/ens/ens';
```
Comment (brad-decker) [inline ui/components/app/modals/account-details-modal/account-details-modal.component.js:11]: [req] unfortunately, ENS is a controller that we should only utilize in the background. Fortunately, though there is an action for this already! tryReverseResolveAddress from the redux actions. For an example of usage, see the transaction-list-item-details.component and .container files.

File: ui/components/app/modals/account-details-modal/account-details-modal.component.js [hunk 2]
```diff
@@ -22,6 +29,39 @@ export default class AccountDetailsModal extends Component {
     trackEvent: PropTypes.func,
   };
 
+  state = {
+    name: this.props.selectedIdentity.name || '',
+  };
+
+  componentDidMount() {
```
Comment (brad-decker) [inline ui/components/app/modals/account-details-modal/account-details-modal.component.js:36]: [req] Rather than extending this component to rely more on class component lifecycle methods, lets convert this component to a functional one. This component is relatively easy to do because it only currently has a render method and no state.

File: ui/components/ui/editable-label/editable-label.js [hunk 3]
```diff
@@ -12,8 +12,18 @@ class EditableLabel extends Component {
   state = {
     isEditing: false,
     value: this.props.defaultValue || '',
+    ensName: '',
   };
 
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (brad-decker) [inline ui/components/ui/editable-label/editable-label.js:18]: [Q] can you explain why these changes are necessary? We should really try to avoid adding more usages of UNSAFE lifecycle methods.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Thanks for the submission and sorry for the delay in response!

**Comment by darkwing** [comment]
@hsy822 We had a few questions with regard to your PR that we'd love answered!  Thank you!

**Comment by hsy822** [comment]
> @hsy822 We had a few questions with regard to your PR that we'd love answered! Thank you!

Sorry for checking so late. I'll check it out within this week.

File: ui/components/app/modals/account-details-modal/account-details-modal.component.js (hunk 1 above)
Comment (hsy822) [reply ui/components/app/modals/account-details-modal/account-details-modal.component.js:13]: Thanks for your advice. I just changed the component to a functional type.
And instead of using tryReverseResolveAddress in account-details-modal.component.js, I use ensResolutionByAddress to get ENS in account-details-modal.container.js. 
Is it okay to do this?

```
account-details-modal.container.js

const mapStateToProps = (state) => {
  const { metamask } = state;
  const { ensResolutionsByAddress } = metamask;
  const selectedIdentity = getSelectedIdentity(state);
  let ensName;
  if (selectedIdentity.address) {
    const address = toChecksumHexAddress(selectedIdentity.address);
    ensName = ensResolutionsByAddress[address] || '';
  }
  return {
    chainId: getCurrentChainId(state),
    selectedIdentity,
    keyrings: state.metamask.keyrings,
    rpcPrefs: getRpcPrefsForCurrentProvider(state),
    ensName,
  };
};
```

File: ui/components/ui/editable-label/editable-label.js (hunk 3 above)
Comment (hsy822) [reply ui/components/ui/editable-label/editable-label.js:18]: If user has ENS, this is to prevent ENS from always being the default value whenever user clicks the edit button. But I am not going to do this way:) 
thanks

File: ui/components/app/modals/account-details-modal/account-details-modal.component.js (hunk 1 above)
Comment (hsy822) [reply ui/components/app/modals/account-details-modal/account-details-modal.component.js:13]: Oh, I need both. I will upload code soon:)

**Comment by brad-decker** [comment]
@hsy822 sorry for letting this slip through the cracks and allowing it to become super stale. I just opened a new pr #15487 that I derived from rebasing your work here onto the latest develop changes. I will see it through to merge. Thank you for your contribution! Your commit was maintained in the rebase so you'll get attribution on the final work.

**Comment by 0xhsy** [comment]
@brad-decker 

I had a hard time because I couldn't speak English well, but I solved it well thanks to you. Thank you very much.

I wanted to contact you, but I couldn't find the contact.

There are many things I want to ask you, and if you are okay with it, please contact me. 

hyunsy822@gmail.com

Regard,
Sooyoung
